terraform/snowflake/modules/elt/roles.tf

```diff
@@ -206,3 +224,61 @@ resource "snowflake_grant_account_role" "transform_read_to_analytics_rwc" {
   role_name        = "${module.transform.name}_READ"
   parent_role_name = "${module.analytics.name}_READWRITECONTROL"
 }
+
+# Grant the Streamlit roles to the Reporter role in the current environment
+# The cross-environment grant of the ${module.raw.name}_${var.environment}_STREAMLIT role
+# to the REPORTER_DEV role will handled outside of this Terraform configuration.
+# via manual SQL execution
+
+resource "snowflake_grant_account_role" "streamlit_raw_to_reporter" {
+  provider         = snowflake.useradmin
+  role_name        = snowflake_account_role.streamlit_raw.name
+  parent_role_name = snowflake_account_role.reporter.name
```

Comment:
Same here, the reporter role shouldn't have any permissions to edit objects in RAW or TRANSFORM